React sidebar mobile version

peterzimon opened the pull request:
ref https://linear.app/ghost/issue/BER-2913/port-sidebar

- Mobile bottom navigation bar and handling responsive sizes for the new React sidebar was completely missing. This PR introduces standard ShadCN sidebar for mobile and a custom mobile navigation bar, similar to the one in Ember.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Introduces a mobile bottom navigation and responsive sidebar for Admin, refactors nav links to use route-aware matching, and adjusts Shade mobile behavior and container heights.
> 
> - **Admin UI (Sidebar & Layout)**
>   - Add `MobileNavBar` and include it in `AdminLayout` footer.
>   - Refactor sidebar into `AppSidebarHeader`, `AppSidebarContent`, `AppSidebarFooter`; move settings/help into new `NavSettings`.
>   - Update nav items to use `to` props and route-aware active state via `useIsActiveLink` (uses `useMatch`).
>   - Improve mobile UX: auto-close sidebar on link click; tweak icon visibility and user menu sizing.
> - **Routing/Framework**
>   - Export `matchPath` and `useMatch` from `react-router` via `admin-x-framework` index for consumers.
> - **Shade (design system)**
>   - Increase mobile sidebar width to `28rem`; make `SidebarTrigger` accept custom children.
>   - Raise mobile breakpoint to 801px via `useIsMobile`.
>   - Add `--mobile-navbar-height` and `.shade.app-container`; adjust mobile heights for Ember host.
> - **Apps (Posts/Stats)**
>   - Apply `.app-container` to `ShadeApp` and adjust `MainLayout` heights to `calc(100svh - var(--mobile-navbar-height))` with desktop override.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b2d9ba39c0d598ba178d7fca6ba2461f866235b1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

peterzimon committed:
Added basic sidebar trigger (WIP)

peterzimon committed:
Static buttons in bottom bar

peterzimon committed:
Factored out components and created hook for isActive

peterzimon committed:
Close sidebar on mobile on navigation

peterzimon committed:
Updated button widths

peterzimon committed:
Updated mobile navbar height

peterzimon committed:
Made post view chevron to show up on mobile

peterzimon committed:
Updated footer

peterzimon committed:
Structure cleanup

peterzimon committed:
Button updates

peterzimon committed:
Updated user menu width for mobile

peterzimon committed:
Setting mobile width to 800px

peterzimon committed:
Cleanup

peterzimon committed:
Switched to fixed position for mobile nav bar

peterzimon committed:
Updated all mobile sidebar breakpoints

peterzimon committed:
Updated hook mobile size

peterzimon committed:
Structured mobile navbar height

peterzimon committed:
Set mobile navbar sticky

peterzimon committed:
Added screen reader tags to mobile menu

jonatansberg reviewed (commented):
I think the code looks good. I added a few nitpicks. I would like to test on a mobile device as well to see how it performs in the real world.

peterzimon committed:
Fixed width and height setting issue with .shade

- We use the .shade class for React portals to scope styles outside the main app container, for example:

```
<div class="shade">
    [Main app]
</div>
...
<div class="shade">[Select dropdown menu]</div>
```

The .shade class had an explicit width and height (100vh) setting which caused these portal components to render an extra screen height resulting in scrolling and overall app container problems.

peterzimon committed:
Fixed stats and posts app main layout for mobile phones

peterzimon committed:
Updated isActiveLink route to use React Router functions

peterzimon committed:
Simplified isActiveLink hook

peterzimon committed:
Cleanup paths

jonatansberg reviewed (approved):
Looks good! 👍 

I added a question about what seemed like a weird class name change.

peterzimon committed:
Updated mobilenavbar & navsettings params
